node/node_newblock.go

```diff
@@ -103,14 +103,25 @@ func (node *Node) WaitForConsensusReadyv2(readySignal chan struct{}, stopChan ch
 						deadline = time.Now().Add(node.BlockPeriod)
 						// Send the new block to Consensus so it can be confirmed.
 						node.BlockChannel <- newBlock
+						utils.GetLogInstance().Debug("hehe", "shardstate", newBlock.Header().ShardState)
 						break
 					}
+
 				}
 			}
 		}
 	}()
 }
 
+func (node *Node) proposeShardStateWithoutBeaconSync(block *types.Block) error {
+	if !core.IsEpochLastBlock(block) {
+		return nil
+	}
+	nextEpoch := new(big.Int).Add(block.Header().Epoch, common.Big1)
+	shardState := core.GetShardState(nextEpoch)
```

Comment:
Please move this line into `(*Node).proposeBeaconShardState`, and use it instead of `CalculateNewShardState` if the epoch is below a certain threshold.  Maybe introduce and use the threshold in the `shardingconfig.Instance` interface:

```go
// ReshardingGenesisEpoch returns the first epoch whose sharding
// assignment is calculated using full resharding facility.
ReshardingGenesisEpoch() *big.Int
```

Then implement this in the three interface implementations.  For mainnet/testnet, you can use a very high number such as `1000000000` to signify that the real cutover epoch is still TBD.

Reply:
As per our discussion. We will still use proposeShardStateWithoutBeaconSync method before beaconSync is finished.